sql/share/errmsg-utf8.txt

```diff
@@ -10786,3 +10786,7 @@ ER_JSON_SCHEMA_KEYWORD_UNSUPPORTED
         eng "%s keyword is not supported"
 ER_EVENTS_NO_ACL
         eng "Event scheduler cannot function with --skip-grant-tables, --bootstrap, or embedded build"
+ER_SERVER_REDIRECT
+        eng "|Server is redirecting clients to '%1$s'|%1$s"
```

Comment:
The use of pipes as a way to make this easy to parse makes it look (IMO) like a hack. What about using a non-localized message like @vaintroub suggested on MDEV-15935? This way the URL is at a constant offset which makes it simpler to deal with. This would eliminate the need to parse the message as the URL would be at a constant offset.

If localization is something that's desired, I think the leading pipe and duplication of data is redundant: you know from the error number that the message will be of some predefined format. You could drop the pipes altogether if you change it to something like `Server is redirecting clients to: %1$s` and make sure the localized message doesn't use colons. This gets rid of the "ugly" error message and also removes the duplication of the URL string. I think it also simplifies the client-side parsing by a little bit:
```c
    char *p= mysql->net.last_error;
    p = strstr(p, ": ");
    if (p) {
        host= p + 2;
        p= strchr(host, ':');
        if (p) {
            *p++ = '\0';
            port= atoi(p);
        }
        else
        {
            /* Restore to the default port, rather than reusing our current one */
            port= 0;
        }
        fprintf(stderr, "Got server redirect to '%s' (port %d)\n", host, port);
        ++server_redirects;
        goto tcp_redirect;
    }
```